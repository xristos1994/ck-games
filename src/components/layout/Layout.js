import React from "react";
import "./layout.css";
import "./colors.css";
import "./fonts.css";
import { Header /* Footer */ } from "./components";

export const Layout = ({ children }) => {
  return (
    <>
      <Header />
      {children}
      {/* <Footer /> */}
    </>
  );
};
