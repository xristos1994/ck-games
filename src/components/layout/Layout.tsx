import React, { FC, ReactElement, ReactChild } from "react";
import { Background, Header, Menu } from "./components";
import "./layout.css";
import "./colors.css";
import "./fonts.css";
import "./button.css";
import "./variables.css";

interface IProps {
  children: ReactChild | ReactChild[];
}

export const Layout: FC<IProps> = ({ children }): ReactElement => {
  return (
    <>
      <Background />
      <Header />
      <Menu />
      {children}
    </>
  );
};
