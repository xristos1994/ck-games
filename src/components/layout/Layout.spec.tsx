import React from 'react';
import renderer from 'react-test-renderer';
import { Layout } from './Layout';

const Background = () => {
  return (
    <div>
      Background
    </div>
  );
};

const Header = () => {
  return (
    <div>
      Header
    </div>
  );
};

const Menu = () => {
  return (
    <div>
      Menu
    </div>
  );
};

jest.mock('./components/', () => ({
  __esModule: true,
  Background: Background,
  Header: Header,
  Menu: Menu
}));

describe('components/Layout', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Layout>
        Test Content
      </Layout>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
