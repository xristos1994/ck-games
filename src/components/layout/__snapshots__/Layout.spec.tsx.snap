// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Layout renders correctly 1`] = `
Array [
  <div>
    Background
  </div>,
  <div>
    Header
  </div>,
  <div>
    Menu
  </div>,
  "Test Content",
]
`;
